pkg/controller/certificates/sync.go

```diff
@@ -104,6 +107,30 @@ func (c *Controller) Sync(ctx context.Context, crt *v1alpha1.Certificate) (err e
 		return nil
 	}
 
+	// check that certificate secret name is unique
+	namespaceCrts := c.certificateLister.Certificates(crtCopy.Namespace)
+	otherCrts, err := namespaceCrts.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	hasDuplicate := false
+	for _, otherCrt := range otherCrts {
+		if otherCrt.Name != crtCopy.Name && otherCrt.Spec.SecretName == crtCopy.Spec.SecretName {
+			hasDuplicate = true
+			break
+		}
+	}
+	if hasDuplicate {
+		c.Recorder.Eventf(crtCopy, corev1.EventTypeWarning, errorDuplicateSecretName, "Duplicate secretName %v", crtCopy.Spec.SecretName)
+		key, err := cache.MetaNamespaceKeyFunc(crtCopy)
+		if err != nil {
+			return nil
```

Comment:
We shouldn't silently ignore errors - if we are going to do it, let's at least log something here